src/Functions/FunctionTokens.h

```diff
@@ -58,6 +60,9 @@ class FunctionTokens : public IFunction
 
 public:
     static constexpr auto name = Generator::name;
+
+    static constexpr bool tokens_maybe_overlapped = std::is_same_v<Generator, URLHierarchy> || std::is_same_v<Generator, URLPathHierarchy>;
```

Comment:
It is weird to mention concrete generator classes (`URLHieararchy`, `URLPathHierarchy`) in the generic function framework class (`FunctionTokens`). Please remove l. 51, 52 and 64 and instead implement `tokens_may_overlap` (note the spelling) as type trait, similar to `static constexpr auto name = "splitByChar"` in src/Functions/splitByChar.cpp which is used in l. 62 of this file.

Reply:
It will not clear to future reader what it means for tokens to "overlap", please document.

Reply:
It is deprecated.